website/docs/docs/build/derived-metrics.md

```diff
@@ -85,17 +86,97 @@ metrics:
+- name: customer_retention
+    description: "Percentage of customers that are active now and those active 1 month ago"
+    label: customer_retention
+    type_params:
+      expr: (active_customers/ active_customers_prev_month)
+      metrics:
+        - name: active_customers
+          alias: current_active_customers
+        - name: active_customers
+          offset_window: 1 month
+          alias: active_customers_prev_month
+```
+
+### Offset windows and granularity
+
+You can query any granularity and offset window combination. The following examples queries a metric with a 7-day offset and a monthly grain:
+
+```yaml
+- name: d7_booking_change
+    description: "Difference between bookings now and 7 days ago"
+    type: derived
+    label: d7 Bookings Change
+    type_params:
+      expr: bookings - bookings_7_days_ago
+      metrics:
+        - name: bookings
+          alias: current_bookings
+        - name: bookings
+          offset_window: 7 days
+          alias: bookings_7_days_ago
+
+```
+
+**Using `offset_to_grain`**
+
+You can set an `offset_to_grain` to specify the granularity or level of detail for the offset, such as a day or hour. Something to note is that you can't use this with `offset_window`. The following example queries a metric with an hourly offset and a monthly grain:
+
+```yaml
+- name: d7_booking_change
+    description: "Calculate bookings per hour for the current month"
+    type: derived
+    label: d7 Bookings Change
+    type_params:
+      expr: bookings * bookings_hourly
+      metrics:
+        - name: bookings
+          alias: current_bookings
+        - name: bookings
+          offset_to_grain: hour
+          alias: bookings_hourly
+```
+
+### Derived metric offset calculation
+
+When you run the query  `mf query --metrics d7_booking_change --group-by metric_time__month` for the metric, here's how it's calculated:
+
+1. We retrieve the raw, unaggregated dataset with the specified measures and dimensions at the smallest level of detail, which is currently 'day'.
+2. Then, we perform an offset join on the daily dataset, followed by performing a date trunc and aggregation to the requested granularity.
+   For example, to calculate `d7_booking_change` for July 2017: 
+   - First we sum up all the booking values for each day in July to calculate the bookings metric.
+   - The following table displays the range of days that make up this monthly aggregation.
+
+|   | Orders | Metric_time |
+| - | ---- | -------- |
+|   | 330 | 2017-07-31 |
+|   | 7030 | 2017-07-30 to 2017-07-02 |
+|   | 78 | 2017-07-01 |
+| Total  | 7438 | 2017-07-01 |
+
+3. Next, we calculate July's bookings with a 7-day offset. The following table displays the range of days that make up this monthly aggregation. Note that the month begins 7 days later (offset by 7 days) on 2017-07-24.
+
+|   | Orders | Metric_time |
+| - | ---- | -------- |
+|   | 329 | 2017-07-24 |
+|   | 6840 | 2017-07-23  to 2017-06-35 |
```

Comment:
hey @Jstein77  - this should be 2017-06-30 right? not 35?

Reply:
Yes! Fat fingered it. Thanks for catching.